src/models/Dialog.js

```diff
@@ -38,4 +38,11 @@ const DialogSchema = new Schema(
   }
 );
 
+DialogSchema.virtual("approval_rate").get(function CalculatesApprovalRate() {
```

Comment:
Import `calculatesApprovalRate` from **src/utils** and use here

```suggestion
DialogSchema.virtual("approval_rate").get(calculatesApprovalRate);
```